src/HasTranslatableSlug.php

```diff
@@ -52,13 +53,58 @@ protected function addSlug()
         });
     }
 
+    protected function generateNonUniqueSlug(): string
+    {
+        $slugField = $this->slugOptions->slugField;
+        $slugString = $this->getSlugSourceString();
+
+        $slug = $this->getTranslations($slugField)[$this->getLocale()] ?? null;
+
+        $hasCustomSlug = $this->hasCustomSlugBeenUsed() && !empty($slug);
+        $hasNonChangedCustomSlug = !$this->slugIsBasedOnTitle() && !empty($slug);
+
+        if($hasCustomSlug || $hasNonChangedCustomSlug){
+            $slugString = $slug;
+        }
+
+        return Str::slug($slugString, $this->slugOptions->slugSeparator, $this->slugOptions->slugLanguage);
+    }
+
+    protected function getOriginalSourceString()
```

Comment:
This function is only used by another function underneath this one. Please move `getOriginalSourceString` underneath the function where it's being used for better readability.